### PR TITLE
RSDK-6040 Add GetProperties endpoint to viam-cartographer

### PR DESCRIPTION
**Pull request by jeremyrhyde**
This PR implements Properties for viam-cartographer.

**Related PR:** https://github.com/viamrobotics/rdk/pull/3376

**JIRA Ticket:** https://viam.atlassian.net/browse/RSDK-6040

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -329,3 +329,5 @@ require (
 	periph.io/x/conn/v3 v3.7.0 // indirect
 	periph.io/x/host/v3 v3.8.2 // indirect
 )
+
+replace go.viam.com/rdk => /Users/jeremyhyde/Development/rdk
```
Comment (jeremyrhyde) [inline go.mod:333]: [note] to be removed after https://github.com/viamrobotics/rdk/pull/3376 and new rdk release

File: viam_cartographer.go [hunk 2]
```diff
@@ -678,3 +689,17 @@ func CheckQuaternionFromClientAlgo(pose spatialmath.Pose, componentReference str
 	}
 	return nil, "", errors.Errorf("error getting SLAM position: quaternion not given, %v", returnedExt)
 }
+
+// checkCloseAndCloudStatus returns an error if the cartographer service has been previously closed or is being run in the cloud.
+func (cartoSvc *CartographerService) checkCloseAndCloudStatus(cmd string) error {
```
Comment (kim-mishra) [inline viam_cartographer.go:694]:
```suggestion
func (cartoSvc *CartographerService) isOpenAndRunningLocally(cmd string) error {
```

File: viam_cartographer.go [hunk 3]
```diff
@@ -604,15 +585,45 @@ func (cartoSvc *CartographerService) LatestMapInfo(ctx context.Context) (time.Ti
...
+			mappingMode = slam.MappingModeLocalizationOnly
+		}
+	}
+
+	prop := slam.Properties{
+		CloudSlam:   cartoSvc.useCloudSlam,
+		MappingMode: mappingMode,
+	}
+	return prop, nil
+}
```
Comment (kim-mishra) [inline viam_cartographer.go:618]:
```suggestion
        props := slam.Properties{
		CloudSlam:   cartoSvc.useCloudSlam,
	}
	
	if cartoSvc.enableMapping && cartoSvc.existingMap == "" {
	    props.MappingMode = slam.MappingModeNewMap
        } else if cartoSvc.enableMapping && cartoSvc.existingMap != "" 
	    props.MappingMode  = slam.MappingModeUpdateExistingMap
        } else if !cartoSvc.enableMapping && cartoSvc.existingMap != "" 
            props.MappingMode  = slam.MappingModeLocalizationOnly
        } else {
	    return slam.Properties{}, errors.New("invalid mode: localizing requires an existing map")
	}

	return prop, nil
}
```

Comment (kim-mishra) [reply viam_cartographer.go:618]: [nit] have a personal preference of removing nesting

File: viam_cartographer.go (hunk 2 above)
Comment (kim-mishra) [reply viam_cartographer.go:694]: [nit] think a name starting with is might read a little cleaner in all of the functions it is used in

**Review by kim-mishra — APPROVED** [review]
2 nits otherwise lgtm

File: viam_cartographer.go (hunk 2 above)
Comment (jeremyrhyde) [reply viam_cartographer.go]: done

**Pull request merged**